src/inc/corcompile.h

```diff
@@ -71,7 +71,7 @@ typedef DPTR(struct CORCOMPILE_VIRTUAL_SECTION_INFO)
 typedef DPTR(struct CORCOMPILE_IMPORT_SECTION)
     PTR_CORCOMPILE_IMPORT_SECTION;
 
-#ifdef _TARGET_X86_
+#if defined(_TARGET_X86_) && defined(WIN32)
```

Comment:
Actually, we should use !defined(FEATURE_PAL) instead of defined(WIN32) everywhere. I have scanned our sources and we don't use WIN32 anywhere except for the PAL tests, which has historical reasons.